homeassistant/components/dsmr/sensor.py

```diff
@@ -374,28 +377,140 @@ class DSMRSensorEntityDescription(SensorEntityDescription):
 )
 
 
-def add_gas_sensor_5B(telegram: dict[str, DSMRObject]) -> DSMRSensorEntityDescription:
-    """Return correct entity for 5B Gas meter."""
-    ref = None
-    if obis_references.BELGIUM_MBUS1_METER_READING2 in telegram:
-        ref = obis_references.BELGIUM_MBUS1_METER_READING2
-    elif obis_references.BELGIUM_MBUS2_METER_READING2 in telegram:
-        ref = obis_references.BELGIUM_MBUS2_METER_READING2
-    elif obis_references.BELGIUM_MBUS3_METER_READING2 in telegram:
-        ref = obis_references.BELGIUM_MBUS3_METER_READING2
-    elif obis_references.BELGIUM_MBUS4_METER_READING2 in telegram:
-        ref = obis_references.BELGIUM_MBUS4_METER_READING2
-    elif ref is None:
-        ref = obis_references.BELGIUM_MBUS1_METER_READING2
-    return DSMRSensorEntityDescription(
-        key="belgium_5min_gas_meter_reading",
-        translation_key="gas_meter_reading",
-        obis_reference=ref,
-        dsmr_versions={"5B"},
-        is_gas=True,
-        device_class=SensorDeviceClass.GAS,
-        state_class=SensorStateClass.TOTAL_INCREASING,
-    )
+def create_mbus_entity(
+    mbus: int, mtype: int, telegram: dict[str, DSMRObject]
+) -> DSMRSensorEntityDescription | None:
+    """Create a new MBUS Entity."""
+    if (
+        mtype == 3
+        and (
+            obis_reference := getattr(
+                obis_references, f"BELGIUM_MBUS{mbus}_METER_READING2"
+            )
+        )
+        in telegram
+    ):
+        return DSMRSensorEntityDescription(
+            key=f"mbus{mbus}_gas_reading",
+            name=f"Gas consumption mbus{mbus}",
+            obis_reference=obis_reference,
+            is_gas=True,
+            device_class=SensorDeviceClass.GAS,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        )
+    if (
+        mtype == 7
+        and (
+            obis_reference := getattr(
+                obis_references, f"BELGIUM_MBUS{mbus}_METER_READING1"
+            )
+        )
+        in telegram
+    ):
+        return DSMRSensorEntityDescription(
+            key=f"mbus{mbus}_water_reading",
+            name=f"Water consumption mbus{mbus}",
+            obis_reference=obis_reference,
+            is_water=True,
+            device_class=SensorDeviceClass.WATER,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        )
```

Comment:
Same here, please add the missing translation_key to `strings.json`.

```suggestion
        return DSMRSensorEntityDescription(
            key=f"mbus{mbus}_water_reading",
            obis_reference=obis_reference,
            is_water=True,
            device_class=SensorDeviceClass.WATER,
            state_class=SensorStateClass.TOTAL_INCREASING,
            translation_key="water_meter_reading",
        )
```